Wikipedia/Code/TalkPageCoffeeRollViewController.swift

```diff
@@ -0,0 +1,58 @@
+import UIKit
+import WMF
+import CocoaLumberjackSwift
+
+final class TalkPageCoffeeRollViewController: ViewController {
+
+    // MARK: - Properties
+
+    fileprivate let viewModel: TalkPageCoffeeRollViewModel
+
+    var coffeeRollView: TalkPageCoffeeRollView {
+        return view as! TalkPageCoffeeRollView
+    }
+
+    // MARK: - Lifecycle
+
+    override func loadView() {
+        let coffeeRollView = TalkPageCoffeeRollView(theme: theme, viewModel: viewModel, frame: UIScreen.main.bounds)
+        view = coffeeRollView
+        coffeeRollView.configure(viewModel: viewModel)
+        scrollView = coffeeRollView.textView        
+    }
+
+    init(theme: Theme, viewModel: TalkPageCoffeeRollViewModel) {
+        self.viewModel = viewModel
+        super.init(theme: theme)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        coffeeRollView.linkDelegate = self
+
+        navigationItem.title = TalkPageViewController.TalkPageLocalizedStrings.title
+    }
+
+    // MARK: - Themeable
+
+    override func apply(theme: Theme) {
+        super.apply(theme: theme)
+        coffeeRollView.apply(theme: theme)
+    }
+
+}
+
+extension TalkPageCoffeeRollViewController: TalkPageTextViewLinkHandling {
+
+    func tappedLink(_ url: URL, sourceTextView: UITextView) {
+        guard let url = URL(string: url.absoluteString, relativeTo: viewModel.talkPageURL) else {
```

Comment:
I'm starting to think this relative link may not be entirely correct here and in the other spot. Links to talk page archives from the coffee roll (which open natively! 😎) causes _their_ links to get screwy. But I think hashing that out is beyond the scope of this task. I'll make a bug note of it.